### PR TITLE
Rewards 2.5 phase 4

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/27882

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/resources/shared/components/vbat_notice.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  onClose?: () => void
+}
+
+export function VBATNotice (props: Props) {
+  const { getString } = React.useContext(LocaleContext)
+  if (!props.vbatDeadline) {
+    return null
+  }
+
+  const deadline = new Date(props.vbatDeadline)
```
Comment (emerick) [inline components/brave_rewards/resources/shared/components/vbat_notice.tsx:50]: Could this be moved inside `getMessages`? Maybe it doesn't matter too much...

File: components/brave_rewards/resources/rewards_panel/components/vbat_notice_modal.style.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (emerick) [inline components/brave_rewards/resources/rewards_panel/components/vbat_notice_modal.style.ts:1]: Nit: Maybe we could update these to 2023.

**Review by emerick — APPROVED** [review]
Looks awesome, just had two really minor nits.

File: components/brave_rewards/resources/shared/components/vbat_notice.tsx (hunk 1 above)
Comment (zenparsing) [reply components/brave_rewards/resources/shared/components/vbat_notice.tsx]: Done.

File: components/brave_rewards/resources/rewards_panel/components/vbat_notice_modal.style.ts (hunk 2 above)
Comment (zenparsing) [reply components/brave_rewards/resources/rewards_panel/components/vbat_notice_modal.style.ts]: Done.

File: browser/extensions/api/brave_rewards_api.cc [hunk 3]
```diff
@@ -644,6 +645,12 @@ void BraveRewardsGetRewardsParametersFunction::OnGetRewardsParameters(
   }
   data.Set("walletProviderRegions", std::move(provider_regions));
 
+  if (!parameters->vbat_deadline.is_null()) {
+    data.Set("vbatDeadline", floor(parameters->vbat_deadline.ToDoubleT() *
```
Comment (tmancey) [inline browser/extensions/api/brave_rewards_api.cc:649]: nit: it is recommended when `base::Value` to use the helpers in `//base/json/values_util.h` for serialization

Comment (tmancey) [reply browser/extensions/api/brave_rewards_api.cc:649]: See TimeToValue and ValueToTime

**Pull request merged**